web/packages/plugins/shared/common/src/components/Pickers/BlueprintPicker.tsx

```diff
@@ -1,114 +1,43 @@
-  }
-`
-
-export const Selector = ({ setShowModal, onChange, blueprintFilter }: any) => {
-  // @ts-ignore-line
-  const { token } = useContext(AuthContext)
-  const dmssApi = new DmssAPI(token)
-  const dataSources: IDataSources = useDataSources(dmssApi)
-  const application = useContext(ApplicationContext)
-
-  const onSelect = (value: string) => {
-    setShowModal(false)
-    onChange(value)
-  }
-
-  return (
-    <div />
-    // <Tree
-    //   state={index.models.tree.models.tree}
-    //   operations={index.models.tree.operations}
-    // >
-    //   {(renderProps: TreeNodeRenderProps) => {
-    //     const { actions, nodeData } = renderProps
-    //
-    //     if (nodeData.meta.type === blueprintFilter) {
-    //       const onClick = () => {
-    //         onSelect(`${renderProps.path}/${nodeData.title}`)
-    //       }
-    //       return (
-    //         <div
-    //           style={{ display: 'flex', flexDirection: 'row' }}
-    //           onClick={onClick}
-    //         >
-    //           {renderProps.iconGroup(() => onClick())}
-    //           {nodeData.title}
-    //         </div>
-    //       )
-    //     } else {
-    //       return (
-    //         <div
-    //           onClick={() => handleOpenOrExpand(renderProps)}
-    //           style={{ display: 'flex', flexDirection: 'row' }}
-    //         >
-    //           {renderProps.iconGroup(() => handleOpenOrExpand(renderProps))}
-    //           {nodeData.title}
-    //           {nodeData.isLoading && (
-    //             <small style={{ paddingLeft: '15px' }}>Loading...</small>
-    //           )}
-    //         </div>
-    //       )
-    //     }
-    //   }}
-    // </Tree>
-  )
-}
-
-export const BlueprintPicker = (props: BlueprintPickerProps) => {
+export const BlueprintPicker = (props: {
+  onChange: (type: string) => void
+  formData: string
+  blueprintFilter?: string
+}) => {
   const {
     onChange,
     formData,
-    uiSchema,
     blueprintFilter = BlueprintEnum.BLUEPRINT,
   } = props
   const [showModal, setShowModal] = useState<boolean>(false)
-  const selectorProps = { setShowModal, onChange, blueprintFilter }
 
   return (
-    <>
-      <div style={{ width: '100%' }}>
-        <Input
-          style={{ width: '280px', margin: '0 8px', cursor: 'pointer' }}
-          type="string"
-          value={formData}
-          placeholder="Select"
-          onClick={() => setShowModal(true)}
+    <div style={{ width: '100%' }}>
+      <Input
+        style={{ width: '280px', margin: '0 8px', cursor: 'pointer' }}
+        type="string"
+        value={formData}
+        placeholder="Select"
+        onClick={() => setShowModal(true)}
+      />
+      <Modal
+        toggle={() => setShowModal(!showModal)}
+        open={showModal}
+        title={'Select a blueprint as type'}
+      >
+        <TreeView
+          onSelect={(node: TreeNode) => {
+            if (node.type !== blueprintFilter) return // Only allowed to select blueprints
```

Comment:
if (node.type !== BlueprintEnum.BLUEPRINT)

should probably give feedback to user why it does not work to select something that is not a blueprint, for example add NotificationManager.error('You can only select a blueprint') before the return statement

Reply:
done